src/SonarScanner.MSBuild.Common/Interfaces/IPlatformHelper.cs

```diff
@@ -24,8 +24,15 @@ namespace SonarScanner.MSBuild.Common;
 
 public interface IPlatformHelper
 {
+    OS OperatingSystem { get; }
     string GetFolderPath(Environment.SpecialFolder folder, Environment.SpecialFolderOption option);
     bool DirectoryExists(string path);
-    bool IsMacOSX();
-    bool IsWindows();
+
+    public enum OS
```

Comment:
As for coding styles conventions on class members ([ref](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#class-members)), nested `enum` declarations should be placed before properties and methods.

Reply:
I'm personally not a big fan of an enum nested in an interface. You should consider moving it out side the interface but keep it in the same file (separate files seem overload here). If `Os` is to short for you you may name it `PlatformOS`. This is an optional change request.